code/datums/progressbar.dm

```diff
@@ -97,9 +97,8 @@
 			animate(progress_bar.bar, pixel_y = dist_to_travel, time = PROGRESSBAR_ANIMATION_TIME, easing = SINE_EASING)
 
 		LAZYREMOVEASSOC(user.progressbars, target, src)
-		user = null
 
-	if(show_target && (target != user))
+	if(user && show_target && (target != user))
```

Comment:
you can just shift this block under the if(user) instead of doing second user check after the first one